Photino.Native/Photino.Windows.DpiHelp.cpp

```diff
@@ -0,0 +1,83 @@
+#include "Photino.Windows.DpiHelp.h"
+
+// =============================================
+//
+// Types
+//
+// ===================
+typedef UINT(WINAPI *_GetDpiForWindowFunc)(HWND hwnd);
+typedef DPI_AWARENESS_CONTEXT(WINAPI *_SetThreadDpiAwarenessContextFunc)(DPI_AWARENESS_CONTEXT dpiContext);
+typedef int(WINAPI *_GetSystemMetricsForDpiFunc)(int nIndex, UINT dpi);
+
+// =============================================
+//
+// Statics
+//
+// ===================
+static HMODULE s_user32 = nullptr;
+static bool s_initialized = false;
+
+static _GetDpiForWindowFunc _getDpiForWindow;
+static _GetSystemMetricsForDpiFunc _getSystemMetricsForDpi;
+static _SetThreadDpiAwarenessContextFunc _setThreadDpiAwarenessContext;
+
+// =============================================
+//
+// Exports
+//
+// ===================
+void CloseDpiHelper()
+{
+	if (s_user32)
+	{
+		s_initialized = false;
+		FreeLibrary(s_user32);
+	}
+}
+
+int GetScreenHeight(int dpi)
+{
+	if (s_initialized && _getSystemMetricsForDpi != nullptr)
```

Comment:
Should not need to check s_initialized here (and ditto other places too). What I would really suggest, though is, that you have stub functions that do nothing, and that way you can avoid all checks. Then you just need to initialize those functions statically at startup (and reset them in CloseDpiHelper):

`
static _GetDpiForWindowFunc _getDpiForWindow = Fake_getDpiForWindow;
`

Reply:
Great tip! Any reason not to make the "stubs" just a helper function that calls the older api? Here's a pass that does that and gets rid of all the runtime checking. Thanks Pat!